### PR TITLE
Spotify Desk Thing

### PR DESCRIPTION
**Pull request by Peiprjs**
## Describe your project in 1-3 sentences

I want to build a device that shows what song is playing through the speaker that's on top of it.
## Hack Club Slack handle

@Peipr

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Comment by Peiprjs** [comment]
I will do some "engineering" "drawings" later today

File: peiprjs.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+4. Run the web interface on the local device, connect the speaker and voilà! A ~~car~~ desk thing.
+
+## Bill of materials
+
+| Product          | Supplier/Link                       | Cost*   |
+| ---------------  | ----------------------------------  | ------  |
+| Orange Pi Zero 2 | https://www.amazon.es/dp/B09M8N614J | $49.12  |
+| Touchscreen      | https://www.amazon.es/dp/B07PHFQF7W | $56.43  |
+| PLA filament     | https://www.amazon.es/dp/B07R6PL63K | $23.42  |
+| Electric plug    | https://www.amazon.es/dp/B06XXNPLJ9 | $ 9.56  |
```
Comment (garyhtou) [inline peiprjs.md:28]: The Orange Pi Zero 2 is powered by USB C. The cable that is included in this product is Micro USB. Do you already have a USB C cable that you plan on using?

**Review by garyhtou — CHANGES_REQUESTED** [review]
Hey @Peiprjs, I love this idea! I have a friend who has the Spotify car thing and it's really cool. Can't wait to see it be turned into a ~~car~~ desk thing!

I wanted to double-check that you have access to a 3D printer. Please let me know if this is the case, or if you plan to buy one and will need to work it into your budget.

Please also see my comment about the USB cable!

File: peiprjs.md (hunk 1 above)
Comment (Peiprjs) [reply peiprjs.md:28]: Yes I do.

**Comment by Peiprjs** [comment]
I do indeed have access to a 3D printer and a USB-C cable.

**Review by garyhtou — CHANGES_REQUESTED** [review]
Will you be needing any screws and/or nuts to assemble your 3d printed parts and mount the Orange Pi? Also, I'd love to hear a little more regarding your plans for building the web interface!

**Comment by Peiprjs** [comment]
No, the screen already has the screws that I will need. 
And regarding the web interface, my plan was to design it using ReactJS. I've used it before in a Hackathon. Then, I'd figure out a way that, on poweron the Orange Pi to launch a website. I think I know how to do it though.

**Comment by Peiprjs** [comment]
I'll either use ReactJS or just make a plain HTML website with some JavaScript magic in the back.

**Comment by Peiprjs** [comment]
image
Here is a very rough sketch

**Review by garyhtou — APPROVED** [review]
Sounds good. Thanks for adding the additional comments @Peiprjs! You're all good to go now.

a member of @hackclub/bank-ops will be reaching out with next steps :). I can't wait to see this project come to completion!

**Comment by Peiprjs** [comment]
Yay! Thank you very much! My timezone is CET, but my sleep schedule is so good it lines up with american office hours.

**Review by daisybanks — APPROVED** [review]
Hi @Peiprjs! This sounds awesome and like something I'd totally use! I'm dming you on Slack now so we can talk about how to get you your grant woo woo! 🥳

**Pull request merged**